src/mlpack/core/data/load_impl.hpp

```diff
@@ -393,6 +406,187 @@ bool Load(const std::string& filename,
   return true;
 }
 
+// For loading data into sparse matrix
+template <typename eT>
+bool Load(const std::string& filename,
+          arma::SpMat<eT>& matrix,
+          const bool fatal,
+          const bool transpose)
+{
+  Timer::Start("loading_data");
+
+  // Get the extension.
+  std::string extension = Extension(filename);
+
+  // Catch nonexistent files by opening the stream ourselves.
+  std::fstream stream;
+#ifdef  _WIN32 // Always open in binary mode on Windows.
+  stream.open(filename.c_str(), std::fstream::in | std::fstream::binary);
+#else
+  stream.open(filename.c_str(), std::fstream::in);
+#endif
+  if (!stream.is_open())
+  {
+    Timer::Stop("loading_data");
+    if (fatal)
+      Log::Fatal << "Cannot open file '" << filename << "'. " << std::endl;
+    else
+      Log::Warn << "Cannot open file '" << filename << "'; load failed."
+          << std::endl;
+
+    return false;
+  }
+
+  bool unknownType = false;
+  arma::file_type loadType;
+  std::string stringType;
+
+  // a temporary file to handle CSV.
+  std::string tempFileName = "temp.txt";
+
+  // if is it a csv,
+  // the target format of the type: arma::coord_ascii expects
+  // values without commas
+  // so we remove the commas and store them in a tempFileName
+  if (extension == "csv")
+  {
+    // Temporary CSV stream to store the values without commas
+    std::ofstream csvstream(tempFileName);
+    if ( !csvstream.is_open() )
+    {
+      if (fatal)
+      {
+        Log::Fatal << "Unable to create temporary file for processing."
+            << std::endl;
+      }
+      else
+      {
+        Log::Warn << "Unable to create temporary file for processing."
+            << std::endl;
+      }
+
+      return false;
+    }
+
+    // iterators to perform the comma with space replacement
+    std::istreambuf_iterator<char> in_itr(stream);
+    std::ostreambuf_iterator<char> out_itr(csvstream);
+
+    // replacing comma with a space and storing to csvstream
+    std::replace_copy(in_itr,
+                      std::istreambuf_iterator<char>(),
+                      out_itr,
+                      ',',
+                      ' ');
```

Comment:
You're right, `coord_ascii` doesn't work when there are commas.  So, therefore, I think that we should only support `.txt` and remove this code (sorry about that, I know that you put time into it).  If Armadillo later adds support for commas, then we can add `.csv` here too.

Reply:
So could we use CSV to support broad sparse matrices ?

Reply:
No, we can't use arma::csv_ascii as of now.

(I checked the current arma version ) (ARMA version: 8.400.0 (Entropy Bandit))

Currently, only arma::coord_ascii and arma::arma_binary are supported in SpMat_meat.hpp load() function. 

So you are right, only txt should be supported, 
But, tsv files having \\t can also be supported. So keeping txt, tsv and bin.

Reply:
Yep, I agree with your assessment here.  Maybe later we will be able to add formats but this is good for now. :+1: